client.go

```diff
@@ -356,8 +356,13 @@ func (c *Client) Auth(a sasl.Client) error {
 	if err != nil {
 		return err
 	}
-	resp64 := make([]byte, encoding.EncodedLen(len(resp)))
-	encoding.Encode(resp64, resp)
+	var resp64 []byte
+	if len(resp) > 0 {
+		resp64 = make([]byte, encoding.EncodedLen(len(resp)))
+		encoding.Encode(resp64, resp)
+	} else {
```

Comment:
Oh, one last thing: the go-sasl API has the same nil vs empty slice footgun in the `Client` interface. Namely, here `resp = nil` means no initial response, while `resp = []byte{}` means an empty initial response.

The fix would be to replace this line with `} else if resp != nil {` I believe.

Reply:
Hm, ok. Force pushed update. I haven't tested it though, wasn't immediately sure how to (and no time to investigate more right now).

Reply:
That's OK. Would need to find a SASL mechanism with no initial response, I don't know of any.

It seems like the push hasn't gone through?

Reply:
Ah, sorry. Forgot the commit in between add and push... 🥵

Reply:
Was that change what you intended?